app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,150 @@
+                    )
+                )
+            }
+        }
+    }
+
+    private fun getUpcomingMovies() {
+        viewModelScope.launch {
+            movieRepository.upcomingMovies(MovieCategory.UPCOMING_TODAY).collect {
+                upcomingMoviesCategorySelected.emit(
+                    homeScreenMapper.toHomeMovieCategoryViewState(
+                        movieCategories = upcomingCategoryLabels,
+                        selectedMovieCategory = MovieCategory.UPCOMING_TODAY,
+                        movies = it
+                    )
+                )
+            }
+        }
+    }
+
+    fun onCategoryLabelClick(categoryId: Int) {
+        when (categoryId) {
+            MovieCategory.POPULAR_STREAMING.ordinal,
+            MovieCategory.POPULAR_FORRENT.ordinal,
+            MovieCategory.POPULAR_ONTV.ordinal,
+            MovieCategory.POPULAR_INTHEATHERS.ordinal
+            -> {
+                viewModelScope.launch {
+                    movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                       popularMoviesCategorySelected.emit(
+                           homeScreenMapper.toHomeMovieCategoryViewState(
+                               movieCategories = popularCategoryLabels,
+                                selectedMovieCategory = MovieCategory.getByOrdinal(categoryId)!!,
+                                movies = it
+                           )
+                       )
+                    }
+                }
+            }
+
+            MovieCategory.NOWPLAYING_MOVIES.ordinal,
+            MovieCategory.NOWPLAYING_TV.ordinal
+            -> {
+                viewModelScope.launch {
+                    movieRepository.nowPlayingMovies(MovieCategory.NOWPLAYING_MOVIES).collect {
+                        nowPlayingMoviesCategorySelected.emit(
+                            homeScreenMapper.toHomeMovieCategoryViewState(
+                                movieCategories = nowPlayingCategoryLabels,
+                                selectedMovieCategory = MovieCategory.getByOrdinal(categoryId)!!,
+                                movies = it
+                            )
+                        )
+                    }
+                }
+            }
+
+            MovieCategory.UPCOMING_TODAY.ordinal,
+            MovieCategory.UPCOMING_THISWEEK.ordinal
+            -> {
+                viewModelScope.launch {
+                    movieRepository.upcomingMovies(MovieCategory.UPCOMING_TODAY).collect {
+                        upcomingMoviesCategorySelected.emit(
+                            homeScreenMapper.toHomeMovieCategoryViewState(
+                                movieCategories = upcomingCategoryLabels,
+                                selectedMovieCategory = MovieCategory.getByOrdinal(categoryId)!!,
+                                movies = it
+                            )
+                        )
+                    }
+                }
+            }
+        }
```

Comment:
Here you should just emit selected category to `_x_MoviesCategorySelected` flows, not collect from repositories. Now you're collecting the same thing multiple times, first in `get_x_Movies()` functions and then here.

Reply:
Am I not supposed to collect new list of movies from repository every time user clicks on a different subcategory?